pkgs/development/python-modules/argilla/default.nix

```diff
@@ -0,0 +1,70 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, deprecated
+, rich
+, backoff
+, packaging
+, pydantic
+, typer
+, tqdm
+, wrapt
+, numpy
+, httpx
+, pandas
+, monotonic
+, pytest
+}:
+let
+  pname = "argilla";
+  version = "1.8.0";
+in
+buildPythonPackage {
+  inherit pname version;
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "argilla-io";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-pUfuwA/+fe1VVWyGxEkvSuJLNxw3sHmp8cQZecW8GWY=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace '"rich <= 13.0.1"' '"rich < 14.0.0"' \
+      --replace '"numpy < 1.24.0"' '"numpy < 1.25.0"'
+  '';
+
+  propagatedBuildInputs = [
+    deprecated
+    rich
+    backoff
+    packaging
+    pydantic
+    typer
+    tqdm
+    wrapt
+    numpy
+    pandas
+    httpx
+    monotonic
+  ];
+
+  # tests try to write to the filesystem
+  doCheck = false;
+
+  nativeCheckInputs = [
+    pytest
+  ];
+
+  meta = with lib; {
+    description = "Argilla: the open-source data curation platform for LLMs";
+    homepage = "https://docs.argilla.io/";
```

Comment:
Personally I prefer the URL to the source repository here, because getting to the docs from there is easy, while getting from the docs to GitHub is annoying.

Also add a meta.changelog here.